### PR TITLE
[Merged by Bors] - feat: port CategoryTheory.CommSq

### PR DESCRIPTION
**Pull request by mo271**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by mo271** [comment]
Please check that the change from `reassoc.1` to `reassoc` or to `@[reassoc (attr := simp)]` is correct.

I added some doc string because the linter complaint, but sometimes it seems a bit excessive: it seems a bit much to 
```
/-- The assertion that a square has a `lift_struct`. -/
class HasLift : Prop where
  /-- Square has a `lift_struct`. -/
  exists_lift : Nonempty sq.LiftStruct
#align category_theory.comm_sq.has_lift CategoryTheory.CommSq.HasLift
```
essentially write the same thing twice for the class and for `exists_lift`...

**Comment by mattrobball** [comment]
> Please check that the change from `reassoc.1` to `reassoc` or to `@[reassoc (attr := simp)]` is correct.
> 
> I added some doc string because the linter complaint, but sometimes it seems a bit excessive: it seems a bit much to
> 
> ```
> /-- The assertion that a square has a `lift_struct`. -/
> class HasLift : Prop where
>   /-- Square has a `lift_struct`. -/
>   exists_lift : Nonempty sq.LiftStruct
> #align category_theory.comm_sq.has_lift CategoryTheory.CommSq.HasLift
> ```
> 
> essentially write the same thing twice for the class and for `exists_lift`...

I checked and the reassoc’ed declarations are there

File: Mathlib/CategoryTheory/CommSq.lean [hunk 1]
```diff
@@ -0,0 +1,241 @@
...
+
+end LiftStruct
+
+instance subsingleton_liftStruct_of_epi (sq : CommSq f i p g) [Epi i] :
+    Subsingleton (LiftStruct sq) :=
+  ⟨fun l₁ l₂ => by
+    ext
+    rw [← cancel_epi i]
+    simp only [LiftStruct.fac_left]
+    ⟩
```
Comment (jcommelin) [inline Mathlib/CategoryTheory/CommSq.lean:175]:
```suggestion
    simp only [LiftStruct.fac_left]⟩
```

**Comment by jcommelin** [comment]
Thanks :tada:

bors merge

**Pull request closed without merging**